src/actions.rs

```diff
@@ -0,0 +1,231 @@
+        self.insert(Name(b"U")).start()
+    }
+
+    /// Start writing the `/PO` dictionary. An action that shall be performed
+    /// when the page containing the annotation is opened. Only permissible for
+    /// annotations. PDF 1.5+.
+    pub fn page_open(&mut self) -> Action<'_> {
+        self.insert(Name(b"PO")).start()
+    }
+
+    /// Start writing the `/PC` dictionary. An action that shall be performed
+    /// when the page containing the annotation is closed. Only permissible for
+    /// annotations. PDF 1.5+.
+    pub fn page_close(&mut self) -> Action<'_> {
+        self.insert(Name(b"PV")).start()
+    }
+
+    /// Start writing the `/PV` dictionary. An action that shall be performed
+    /// when the page containing the annotation becomes visible. Only
+    /// permissible for annotations. PDF 1.5+.
+    pub fn page_visible(&mut self) -> Action<'_> {
+        self.insert(Name(b"PV")).start()
+    }
+
+    /// Start writing the `/PI` dictionary. An action that shall be performed
+    /// when the page containing the annotation is no longer visible in the
+    /// conforming reader’s user interface. Only permissible for annotations.
+    /// PDF 1.5+.
+    pub fn page_invisible(&mut self) -> Action<'_> {
+        self.insert(Name(b"PI")).start()
+    }
+}
+
+/// Only permissible for [widget](crate::types::AnnotationType::Widget)
+/// [annotations](Annotation).
+impl<'a> AdditionalActions<'a> {
+    /// Start writing the `/Fo` dictionary. An action that shall be performed
+    /// when the annotation receives the input focus.
+    /// Only permissible for widget annotations. PDF 1.2+.
+    pub fn focus(&mut self) -> Action<'_> {
+        self.insert(Name(b"Fo")).start()
+    }
+
+    /// Start writing the `/Bl` dictionary. An action that shall be performed
+    /// when the annotation loses the input focus. Only permissible for widget
+    /// annotations. PDF 1.2+.
+    pub fn focus_loss(&mut self) -> Action<'_> {
+        self.insert(Name(b"Bl")).start()
+    }
+}
+
+/// Only permissible for [page objects](Page).
+impl<'a> AdditionalActions<'a> {
+    /// Start writing the `/O` dictionary. This sets the action that shall be
+    /// performed when the page is opened. This action is independent of any
+    /// that may be defined by the open action entry in the
+    /// [document catalog](Catalog) and shall be executed after such an action.
+    /// Only permissible for [page objects](Page). PDF 1.2+.
+    pub fn open(&mut self) -> Action<'_> {
+        self.insert(Name(b"O")).start()
+    }
+
+    /// Start writing the `/C` dictionary. This sets the action that shall
+    /// be performed when the page is closed. This action applies to the page
+    /// being closed and shall be executed before any other page is opened. Only
+    /// permissible for [page objects](Page). PDF 1.2+.
+    pub fn close(&mut self) -> Action<'_> {
+        self.insert(Name(b"C")).start()
+    }
+}
```

Comment:
I opted for `open`/`close` here and `page_open` and `page_close` up in L130-L142, I'm not entirely sure about this, I also thought about adding a prefix to each specific method (`annot`, `widget`, `page` and `cat`). So that these become `page_open` `page_close` and those for annots become `annot_page_open` and `annot_page_close` respectively.

Neither seemed satisfactory to me.